src/common/cluster_set.cc

```diff
@@ -442,19 +442,33 @@ void ClusterSet::ScheduleAlignments(AllAllExecutor* executor) {
   std::cout << "Avoided " << num_avoided << " alignments.\n";
 }
 
-void ClusterSet::DumpJson(const std::string& filename) const {
-  nlohmann::json j = json::array();
-  long long int counter = 0;
+
+
+
+//Add by akash
+void ClusterSet::DumpJson(const std::string& filename, std::vector <std::string> & datasetsFileName) const {
+  
+  nlohmann::json j;
+  long long int counter=0;
+
+  j["datasets"] = json::array();
+  for(long long int c = 0; c < datasetsFileName.size(); c++){
+      j["datasets"].push_back(datasetsFileName[c]);  
+  }
+ 
+  j["clusters"] = json::array();
 
   for (const auto& c : clusters_) {
-    j.push_back(json::array());
+
+    j["clusters"].push_back(json::array());
 
     for (const auto& s : c.Sequences()) {
       nlohmann::json j_temp = json::object();
       j_temp["Genome"] = s.Genome();
       j_temp["Index"] = s.GenomeIndex();
-      j_temp["AbsoluteIndex"] = s.ID();
-      j[counter].push_back(j_temp);
+      j_temp["AbsoluteIndex"] = s.ID(); 
+      j["clusters"][counter].push_back(j_temp);
```

Comment:
why is counter necessary? can't we build the array and then push it onto `j["clusters"]` ?